util/dockerfiles/devcontainer/Dockerfile

```
@@ -39,15 +40,60 @@
 # In the final stage this this is copied into /usr/local/bin/gem5. This ensures
 # there is a pre-built gem5 binary in each devcontainer. This can save time
 # if the container is used for education or demonstration purposes.
-FROM --platform=${BUILDPLATFORM} ghcr.io/gem5/ubuntu-24.04_all-dependencies as builder
-RUN git clone --branch stable https://github.com/gem5/gem5
+FROM ghcr.io/gem5/ubuntu-22.04_all-dependencies:latest as builder
+RUN git clone --branch release-staging-v24.0.0.0 https://github.com/gem5/gem5
 WORKDIR /gem5
 RUN scons build/ALL/gem5.opt -j`nproc`
+RUN scons defconfig build/ALL_CHI build_opts/ALL && \
+    scons setconfig build/ALL_CHI RUBY_PROTOCOL_CHI=y && \
+    scons build/ALL_CHI/gem5.opt -j`nproc`
+RUN scons build/VEGA_X86/gem5.opt -j`nproc`
 
 # Stage 2: The final stage where we create the devcontainer image.
 # This includes all dependencies for building and running gem5, the
 # dependencies for developing gem5, and a pre-built gem5 binary.
-FROM --platform=${BUILDPLATFORM} ghcr.io/gem5/ubuntu-24.04_all-dependencies
+FROM ghcr.io/gem5/ubuntu-22.04_all-dependencies:latest
 RUN apt -y update  && \
     apt -y install vim
-COPY --from=builder /gem5/build/ALL/gem5.opt /usr/local/bin/gem5
+    COPY --from=builder /gem5/build/ALL/gem5.opt /usr/local/bin/gem5-default
+    COPY --from=builder /gem5/build/ALL_CHI/gem5.opt /usr/local/bin/gem5
+    COPY --from=builder /gem5/build/VEGA_X86/gem5.opt /usr/local/bin/gem5-vega
```

There is a weird leading whitespaces here.

```suggestion
COPY --from=builder /gem5/build/ALL/gem5.opt /usr/local/bin/gem5-default
COPY --from=builder /gem5/build/ALL_CHI/gem5.opt /usr/local/bin/gem5
COPY --from=builder /gem5/build/VEGA_X86/gem5.opt /usr/local/bin/gem5-vega
```